### PR TITLE
Be cool prometheus...just be cool

### PR DESCRIPTION
**Pull request by derekpierre**
Related to #2373 

Reduce the greediness of prometheus metrics collection.

On average prometheus metric collection performs ~40 requests/10s 😱 

- [X] Reduce frequency
- [X] Remove WorkLock event metrics collections.

### CODE REVIEW

File: docs/source/guides/network_node/ursula_configuration_guide.rst [hunk 1]
```diff
@@ -218,6 +218,10 @@ parameters to the ``nucypher ursula run`` command:
 The corresponding endpoint, ``http://<node_ip>:<METRICS PORT>/metrics``, can be used as a Prometheus data source for
 monitoring including the creation of alert criteria.
 
+By default metrics will be collected every 90 seconds but this can be modified using the ``--metrics-interval`` option.
```
Comment (KPrasch) [inline docs/source/guides/network_node/ursula_configuration_guide.rst:221]: :boom:

File: nucypher/utilities/prometheus/collector.py [hunk 2]
```diff
@@ -432,65 +432,3 @@ def _event_occurred(self, event) -> None:
         super()._event_occurred(event)
         self.metrics["current_worker_is_me_gauge"].set(
             self.contract_agent.get_worker_from_staker(self.staker_address) == self.worker_address)
-
-
-class BidRefundCompositeEventMetricsCollector(MetricsCollector):
-    """
-    Collector for both Bid and Refund WorkLock events.
-
-    Both Bid and Refund events additionally update the same metric of how much eth is deposited by the staker
```
Comment (cygnusv) [inline nucypher/utilities/prometheus/collector.py:441]: I agree Bid is useless now, but Refund will be useful for a while

**Comment by jMyles** [comment]
One thing I'm not clear on - are we thinking that this is a substantial part of #2373?  How many nodes are actually using the prometheus integration?

File: nucypher/utilities/prometheus/collector.py (hunk 2 above)
Comment (derekpierre) [reply nucypher/utilities/prometheus/collector.py:441]: ✔️

**Comment by derekpierre** [comment]
> How many nodes are actually using the prometheus integration?

@jMyles I'm unsure of that proportion of the network, but even if it is a small portion of actual nodes, prometheus is definitely one factor in higher request counts for those who use it. Note that this issue doesn't claim to fix #2373, just related to it.

File: nucypher/utilities/prometheus/collector.py [hunk 3]
```diff
@@ -434,63 +433,19 @@ def _event_occurred(self, event) -> None:
...
+        self.metrics["worklock_deposited_eth_gauge"] = Gauge(f'{metrics_prefix}_worklock_current_deposited_eth',
+                                                             'Worklock deposited ETH',
+                                                             registry=registry)
 
-    def collect(self) -> None:
-        for collector in self.collectors:
-            collector.collect()
+    def _event_occurred(self, event) -> None:
+        super()._event_occurred(event)
+        self.metrics["worklock_deposited_eth_gauge"].set(self.contract_agent.get_deposited_eth(self.staker_address))
```
Comment (vzotova) [inline nucypher/utilities/prometheus/collector.py:451]: Sorry for dumb question: I see name of class includes "refund" and here "deposited", could you explain a little bit?

Comment (derekpierre) [reply nucypher/utilities/prometheus/collector.py:451]: Not dumb at all. Original code from P2O - both Bid and Refund events were tracked and when either event occurred, in addition to event information, they also updated the common metric `worklock_deposited_eth_gauge` - the thought was that bid increased the deposited ETH and refund decreased it. The naming could be improved - but that would involve a change to the dashboard as well.

Comment (vzotova) [reply nucypher/utilities/prometheus/collector.py:451]: Still do not understand everything, I mean bid is removed, and name contains refund but metrics for deposit. Anyway if it's on purpose then ok

Comment (KPrasch) [reply nucypher/utilities/prometheus/collector.py:451]: I agree this reads a bit confusing - why set `contract_agent.get_deposited_eth` for the *Refund* event?

Comment (derekpierre) [reply nucypher/utilities/prometheus/collector.py:451]: When a refund occurs for WorkLock the following also happens:
```python
info.depositedETH = info.depositedETH.sub(refundETH);
```
And I think this is what is attempted to be captured.

**Review by vzotova — APPROVED** [review]

**Review by KPrasch — APPROVED** [review]
:cowboy_hat_face:

**Pull request merged**